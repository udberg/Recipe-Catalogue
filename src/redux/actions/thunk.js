import { fetchMeals, fetchMeal, fetchSearchMeals } from '../../API/api';
import {
  loadingMeal, errorMeal, successMeal, loadingMeals, errorMeals, successMeals,
} from './index';

const fetchMealsByCategory = (category) => async (dispatch) => {
  dispatch(loadingMeals());
  const response = fetchMeals(category);
  try {
    const data = await response;
    return dispatch(successMeals(data.meals));
  } catch (e) {
    return dispatch(errorMeals('Error while fetching data.'));
  }
};

const searchMeals = (query) => async (dispatch) => {
  dispatch(loadingMeals());
  const response = fetchSearchMeals(query);
  try {
    const data = await response;
    if (data.meals) {
      return dispatch(successMeals(data.meals));
    }
    return dispatch(errorMeals(`There is no meal with name: ${query}`));
  } catch (e) {
    return dispatch(errorMeals('Error while fetching data.'));
  }
};

const fetchMealById = (id) => async (dispatch) => {
  dispatch(loadingMeal());
  const response = fetchMeal(id);
  try {
    const data = await response;
    return dispatch(successMeal(data.meals[0]));
  } catch (e) {
    return dispatch(errorMeal('Error while fetching data.'));
  }
};

export { fetchMealsByCategory, searchMeals, fetchMealById };
