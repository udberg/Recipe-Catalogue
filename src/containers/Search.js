import classNames from 'classnames';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import { searchMeals } from '../redux/actions/thunk';
import styles from '../styles/Search.module.css';

const cx = classNames;

const Search = ({ searchMeals }) => {
  const history = useHistory();

  const handleSearch = (event) => {
    event.preventDefault();
    const { search: { value } } = event.target;
    searchMeals(value);
    event.target.reset();
    history.push('/');
  };

  return (
    <form onSubmit={handleSearch}>
      <input className={cx(styles['search-input'])} type="search" placeholder="Search your recipes" name="search" />
    </form>
  );
};

Search.propTypes = {
  searchMeals: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  searchMeals: (query) => dispatch(searchMeals(query)),
});

export default connect(null, mapDispatchToProps)(Search);
