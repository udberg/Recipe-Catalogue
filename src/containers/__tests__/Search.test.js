import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import ConfigureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Search from '../Search';

const mockStore = ConfigureMockStore([thunk]);
jest.mock('../../API/api');

let store;
beforeEach(() => {
  store = mockStore({});
});

test('should test for search component snapshot', () => {
  const { container } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Search />
      </MemoryRouter>
    </Provider>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
