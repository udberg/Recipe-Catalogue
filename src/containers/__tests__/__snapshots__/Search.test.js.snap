// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test for search component snapshot 1`] = `
<form>
  <input
    class="search-input"
    name="search"
    placeholder="Search your recipes"
    type="search"
  />
</form>
`;
